src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs

```diff
@@ -417,6 +418,51 @@ static IEnumerable<string> TestStrings(bool generateLongTestCase)
             AnalyzerAssert.Valid<TestCaseSourceUsesStringAnalyzer>(testCode);
         }
 
+        [Test]
+        public void AnalyzeWhenAllInformationIsProvidedByAttributeWhenDataClassIsInSeparateAssembly()
+        {
+            var testCode = @"
+    using NUnit.Framework;
+    using Project2;
+
+    namespace Project1
+    {
+        public class Tests
+        {
+            [Test]
+            [TestCaseSource(typeof(TestData), nameof(TestData.TestCaseDataProvider))]
+            public void Test1(int a, int b, int c)
+            {
+                Assert.That(a + b, Is.EqualTo(c));
+            }
+        }
+    }";
+
+            var testDataCode = @"
+    using System.Collections;
+
+    namespace Project2
+    {
+        public static class TestData
+        {
+            public static IEnumerable TestCaseDataProvider()
+            {
+                yield return new object[] { 1, 2, 3 };
+                yield return new object[] { 2, 3, 5 };
+            }
+        }
+    }";
+
+            var testDataReference = MetadataReferences.CreateBinary(testDataCode);
+            var references = AnalyzerAssert.MetadataReferences.Concat(new[] { testDataReference });
```

Comment:
```suggestion
            var references = AnalyzerAssert.MetadataReferences.Append(testDataReference);
```

Reply:
It's not available for net461, which is one of our test targets (I'm not sure why though)